sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2515,6 +2515,78 @@ ProgramManager::getOrCreateKernel(const context &Context,
                          BuildResult->Val.second);
 }
 
+sycl::detail::pi::PiKernel ProgramManager::getCachedMaterializedKernel(
+    const std::string &KernelName,
+    const std::vector<unsigned char> &SpecializationConsts) {
+  if (DbgProgMgr > 0)
```

Comment:
```suggestion
  if constexpr (DbgProgMgr > 0)
```

I think it's a mistake we don't do it in other places.

Reply:
Not using `constexpr` if clauses in `DbgProgMgr`? I'll go ahead and add them in the whole file.